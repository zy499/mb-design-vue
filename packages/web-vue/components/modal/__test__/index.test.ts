import { nextTick } from 'vue';
import { mount } from '@vue/test-utils';
import Modal from '../index';
import ModalComponent from '../modal.vue';

describe('Modal', () => {
  test('should render modal and emit ok/cancel event', async () => {
    const wrapper = mount(ModalComponent, {
      props: {
        defaultVisible: true,
        renderToBody: false,
      },
      slots: {
        default: `<div>Modal Body</div>`,
        title: 'Title',
      },
    });

    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    const buttons = wrapper.findAll('button');
    await buttons[0].trigger('click');
    expect(wrapper.emitted('cancel')).toHaveLength(1);
    await buttons[1].trigger('click');
    expect(wrapper.emitted('ok')).toHaveLength(1);
  });

  test('should show modal and call cb', async () => {
    const onOk = jest.fn();
    const onCancel = jest.fn();

    const wrapper = mount({
      template: `
        <button @click="handleClick">Click</button>`,
      methods: {
        handleClick() {
          Modal.open({
            title: 'title',
            content: 'content',
            onOk,
            onCancel,
          });
        },
      },
    });

    await wrapper.find('button').trigger('click');
    expect(document.body.outerHTML).toMatchSnapshot();
    const buttons = document.querySelectorAll('button');
    await buttons[0]?.click();
    expect(onCancel).toBeCalled();
    await buttons[1]?.click();
    expect(onOk).toBeCalled();
  });

  test('should render simple modal', async () => {
    const wrapper = mount(ModalComponent, {
      props: {
        defaultVisible: true,
        simple: true,
        renderToBody: false,
      },
      slots: {
        default: `<div>Modal Body</div>`,
        title: 'Title',
      },
    });

    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
