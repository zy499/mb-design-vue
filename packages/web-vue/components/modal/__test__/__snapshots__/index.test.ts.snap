// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render modal and emit ok/cancel event 1`] = `
"<!--teleport start-->
<div class=\\"mb-modal-container\\" style=\\"z-index: 1001;\\">
  <transition-stub>
    <div class=\\"mb-modal-mask\\"></div>
  </transition-stub>
  <div class=\\"mb-modal-wrapper mb-modal-wrapper-align-center\\">
    <transition-stub>
      <div class=\\"mb-modal\\">
        <div class=\\"mb-modal-header\\">
          <div class=\\"mb-modal-title mb-modal-title-align-undefined\\">
            <!--v-if-->Title
          </div>
          <div tabindex=\\"-1\\" role=\\"button\\" aria-label=\\"Close\\" class=\\"mb-modal-close-btn\\"><span class=\\"mb-icon-hover\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-close\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M9.857 9.858 24 24m0 0 14.142 14.142M24 24 38.142 9.858M24 24 9.857 38.142\\"></path></svg></span></div>
        </div>
        <div class=\\"mb-modal-body\\">
          <div>Modal Body</div>
        </div>
        <div class=\\"mb-modal-footer\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
            <!--v-if-->取消
          </button><button class=\\"mb-btn mb-btn-primary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
            <!--v-if-->确定
          </button></div>
      </div>
    </transition-stub>
  </div>
</div>
<!--teleport end-->"
`;

exports[`Modal should render simple modal 1`] = `
"<!--teleport start-->
<div class=\\"mb-modal-container\\" style=\\"z-index: 1001;\\">
  <transition-stub>
    <div class=\\"mb-modal-mask\\"></div>
  </transition-stub>
  <div class=\\"mb-modal-wrapper mb-modal-wrapper-align-center\\">
    <transition-stub>
      <div class=\\"mb-modal mb-modal-simple\\">
        <div class=\\"mb-modal-header\\">
          <div class=\\"mb-modal-title mb-modal-title-align-undefined\\">
            <!--v-if-->Title
          </div>
          <!--v-if-->
        </div>
        <div class=\\"mb-modal-body\\">
          <div>Modal Body</div>
        </div>
        <div class=\\"mb-modal-footer\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
            <!--v-if-->取消
          </button><button class=\\"mb-btn mb-btn-primary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
            <!--v-if-->确定
          </button></div>
      </div>
    </transition-stub>
  </div>
</div>
<!--teleport end-->"
`;

exports[`Modal should show modal and call cb 1`] = `"<body style=\\"overflow: hidden; width: -1024px; box-sizing: border-box;\\"><div class=\\"arco-overlay arco-overlay-modal\\"><!--teleport start--><div class=\\"mb-modal-container\\" style=\\"z-index: 1001;\\"><transition-stub><div class=\\"mb-modal-mask\\"></div></transition-stub><div class=\\"mb-modal-wrapper mb-modal-wrapper-align-center\\"><transition-stub><div class=\\"mb-modal\\"><div class=\\"mb-modal-header\\"><div class=\\"mb-modal-title mb-modal-title-align-undefined\\"><!--v-if-->title</div><div tabindex=\\"-1\\" role=\\"button\\" aria-label=\\"Close\\" class=\\"mb-modal-close-btn\\"><span class=\\"mb-icon-hover\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-close\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M9.857 9.858 24 24m0 0 14.142 14.142M24 24 38.142 9.858M24 24 9.857 38.142\\"></path></svg></span></div></div><div class=\\"mb-modal-body\\">content</div><div class=\\"mb-modal-footer\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\"><!--v-if-->取消</button><button class=\\"mb-btn mb-btn-primary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\"><!--v-if-->确定</button></div></div></transition-stub></div></div><!--teleport end--></div></body>"`;
